src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -674,97 +548,140 @@ private static LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData,
+        {
+            if (loadAllPages)
+            {
+                customViewer.CreateCache();
+            }
 
-                    foreach (var page in viewInfo.Pages)
-                    {
-                        xdoc.Element("Pages")
-                            .Add(new XElement(
-                                "PageData",
-                                new XElement("Number", page.Number),
-                                new XElement("Angle", 0)));
-                    }
+            dynamic viewInfo = customViewer.GetViewer().GetViewInfo(ViewInfoOptions.ForHtmlView());
+            LoadDocumentEntity loadDocumentEntity = new LoadDocumentEntity();
 
-                    xdoc.Save(pagesInfoPath);
-                }
+            if (!Directory.Exists(cachePath))
+            {
+                Directory.CreateDirectory(cachePath);
+            }
 
-                List<string> pagesContent = new List<string>();
+            string pagesInfoPath;
+            TryCreatePagesInfoXml(fileCacheSubFolder, viewInfo, out pagesInfoPath);
 
+            foreach (Page page in viewInfo.Pages)
+            {
+                PageDescriptionEntity pageData = GetPageInfo(page, pagesInfoPath);
                 if (loadAllPages)
                 {
-                    List<MemoryStream> pages = new List<MemoryStream>();
-                    MemoryPageStreamFactory pageStreamFactory = new MemoryPageStreamFactory(pages);
+                    pageData.SetData(GetPageContent(page.Number, documentGuid, cachePath));
+                }
 
-                    if (globalConfiguration.Viewer.GetIsHtmlMode())
-                    {
-                        ViewOptions viewOptions = HtmlViewOptions.ForEmbeddedResources(pageStreamFactory);
-                        viewOptions.SpreadsheetOptions.TextOverflowMode = TextOverflowMode.HideText;
+                loadDocumentEntity.SetPages(pageData);
+            }
 
-                        viewer.View(viewOptions);
-                    }
-                    else
-                    {
-                        PngViewOptions pngViewOptions = new PngViewOptions(pageStreamFactory);
+            loadDocumentEntity.SetGuid(documentGuid);
+            return loadDocumentEntity;
+        }
 
-                        viewer.View(pngViewOptions);
-                    }
+        /// <summary>
+        /// Gets document page info.
+        /// </summary>
+        /// <param name="postedData">Posted data with page number.</param>
+        /// <returns>Document page info.</returns>
+        [HttpPost]
+        [Route("loadDocumentPage")]
+        public HttpResponseMessage GetDocumentPage(PostedDataEntity postedData)
+        {
+            string password = string.Empty;
+            try
+            {
+                string documentGuid = postedData.guid;
+                int pageNumber = postedData.page;
+                password = string.IsNullOrEmpty(postedData.password) ? null : postedData.password;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'password'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933706)